tests/core/test_policies.py

```diff
@@ -205,6 +220,60 @@ def create_policy(self, featurizer, priority):
         p = MappingPolicy()
         return p
 
+    @pytest.fixture(scope="module")
+    def domain_with_mapping(self):
+        return Domain.load(DEFAULT_DOMAIN_PATH_WITH_MAPPING)
+
+    @pytest.fixture
+    def tracker(self, domain_with_mapping):
+        return DialogueStateTracker(UserMessage.DEFAULT_SENDER_ID)
+
+    @pytest.fixture(
+        params=[
+            ("default", "utter_default"),
+            ("greet", "utter_greet"),
+            (USER_INTENT_RESTART, ACTION_RESTART_NAME),
+            (USER_INTENT_BACK, ACTION_BACK_NAME),
+        ]
+    )
+    def intent_mapping(self, request):
+        return request.param
+
+    def test_predict_mapped_action(self, priority, domain_with_mapping, intent_mapping):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+        ]
+
+        assert (
+            self._get_next_action(policy, events, domain_with_mapping)
+            == intent_mapping[1]
+        )
+
+    def test_predict_action_listen(self, priority, domain_with_mapping, intent_mapping):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+            ActionExecuted(intent_mapping[1], policy="MappingPolicy"),
+        ]
+        action_planned = self._get_next_action(policy, events, domain_with_mapping)
+        assert action_planned == ACTION_LISTEN_NAME
+
+    def test_do_not_follow_other_policy(
+        self, priority, domain_with_mapping, intent_mapping
+    ):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+            ActionExecuted(intent_mapping[1], policy="other_policy"),
+        ]
+        tracker = get_tracker(events)
+        scores = policy.predict_action_probabilities(tracker, domain_with_mapping)
+        assert scores == [0] * domain_with_mapping.num_actions
```

Comment:
here we test whether the policy ensemble chooses mapping policy or not, but above we only test whether the mapping policy predicts action listen. Can we in the first one actually tests if mapping policy gets picked in the ensemble?

Reply:
I don't think this tests whether the policy is picked but only what it does if it is picked after an action was predicted by a different policy.
I'm not sure if we can check whether it does get picked. Most likely not in the policy though, that should be a test for the ensemble.